code/modules/mob/mob_movement.dm

```diff
@@ -289,42 +289,46 @@
 	. = ..()
 	if(. || HAS_TRAIT(src, TRAIT_SPACEWALK))
 		return TRUE
-	var/atom/movable/backup = get_spacemove_backup()
+	var/atom/movable/backup = get_spacemove_backup(movement_dir)
 	if(backup)
 		if(istype(backup) && movement_dir && !backup.anchored)
-			if(backup.newtonian_move(turn(movement_dir, 180))) //You're pushing off something movable, so it moves
+			if(backup.newtonian_move(turn(movement_dir, 180), TRUE)) //You're pushing off something movable, so it moves
 				to_chat(src, span_info("You push off of [backup] to propel yourself."))
 		return TRUE
 	return FALSE
 
 /**
- * Find movable atoms? near a mob that are viable for pushing off when moving
+ * Finds a target near a mob that is viable for pushing off when moving.
+ * Takes the intended movement direction as input.
  */
-/mob/get_spacemove_backup()
-	for(var/A in orange(1, get_turf(src)))
-		if(isarea(A))
+/mob/get_spacemove_backup(moving_direction)
+	for(var/atom/pushover as anything in orange(1, get_turf(src)))
+		if(isarea(pushover))
 			continue
-		else if(isturf(A))
-			var/turf/turf = A
+		if(moving_direction == get_dir(src, pushover))
+			continue
```

Comment:
Couldn't this break railings / directional windows in some cases? Specifically moving into one facing 90° from your direction of movement.

Reply:
I don't think so no, could you go into more detail?

Reply:
IIRC you can currently push off of (pull on?) directional windows and railings on adjacent turfs to walk into their turf so long as those railings are facing a direction that will not block that movement. This bit discards any objects in the direction of the mobs movement. As such it will discard such windows and railings, potentially preventing that movement.

Reply:
Oh you must be shitting me

Reply:
![dreamseeker_fw6q9l3smq](https://user-images.githubusercontent.com/58055496/154012511-a9a535de-f54d-474c-8ce9-14c6225f4959.gif)
AHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHH

Reply:
Resolved I hope, range instead of orange, and a pass check seems to have fixed it